File: Kotlin/프로그래머스/1/86491. 최소직사각형/최소직사각형.kt
class Solution {
    fun solution(sizes: Array<IntArray>): Int {
        
        var maxX = sizes[0][0]
        var maxY = sizes[0][1]

        if (maxX < maxY) {
            val temp = maxX
            maxX = maxY
            maxY = temp
        }

        for (i in sizes.indices) {

            val x = sizes[i][0]
            val y = sizes[i][1]

            if (y > x) {
                if (maxX < y) {
                    maxX = y
                    println("change maxX = $maxX")
                }
                if (maxY < x) {
                    maxY = x
                    println("change maxY = $maxY")
                }
                continue
            }

            if (maxX < x) {
                maxX = x
                println("change maxX = $maxX")
            }
            if (maxY < y) {
                maxY = y
                println("change maxY = $maxY")
            }
        }

        return maxX * maxY
    }
}

File: C#/프로그래머스/1/86491. 최소직사각형/최소직사각형.cs
using System;

public class Solution {
    public int solution(int[,] sizes) {
        int answer = 0;
        int x = 0;
        int y = 0;
        
        for(int i = 0; i < sizes.GetLength(0); i++)
        {
            if (sizes[i,0] > sizes[i,1]) 
            {
                int t = sizes[i,0];
                sizes[i,0] = sizes[i,1];
                sizes[i,1] = t;
            }
            
            x = Math.Max(sizes[i,0], x);
            y = Math.Max(sizes[i,1], y);
        }
        
        return x * y;
    }
}